evalml/automl/automl_search.py

```diff
@@ -504,45 +507,23 @@ def search(self, X, y, data_checks="auto", show_iteration_plot=True):
             try:
-                if len(current_batch_pipelines) == 0:
-                    try:
-                        if current_batch_pipeline_scores and np.isnan(np.array(current_batch_pipeline_scores, dtype=float)).all():
-                            raise AutoMLSearchException(f"All pipelines in the current AutoML batch produced a score of np.nan on the primary objective {self.objective}.")
-                        current_batch_pipelines = self._automl_algorithm.next_batch()
-                        current_batch_pipeline_scores = []
-                    except StopIteration:
-                        logger.info('AutoML Algorithm out of recommendations, ending')
-                        break
-                pipeline = current_batch_pipelines.pop(0)
-                parameters = pipeline.parameters
-                logger.debug('Evaluating pipeline {}'.format(pipeline.name))
-                logger.debug('Pipeline parameters: {}'.format(parameters))
-
-                if self.start_iteration_callback:
-                    self.start_iteration_callback(pipeline.__class__, parameters, self)
-                desc = f"{pipeline.name}"
-                if len(desc) > self._MAX_NAME_LEN:
-                    desc = desc[:self._MAX_NAME_LEN - 3] + "..."
-                desc = desc.ljust(self._MAX_NAME_LEN)
-
-                if self.max_batches:
-                    update_pipeline(logger, desc, len(self._results['pipeline_results']) + 1, self.max_iterations, self._start, self._automl_algorithm.batch_number)
-                else:
-                    update_pipeline(logger, desc, len(self._results['pipeline_results']) + 1, self.max_iterations, self._start)
+                if current_batch_pipeline_scores and np.isnan(np.array(current_batch_pipeline_scores, dtype=float)).all():
+                    raise AutoMLSearchException(f"All pipelines in the current AutoML batch produced a score of np.nan on the primary objective {self.objective}.")
+                current_batch_pipelines = self._automl_algorithm.next_batch()
+                current_batch_pipeline_scores = []
+            except StopIteration:
+                logger.info('AutoML Algorithm out of recommendations, ending')
+                break
 
-                evaluation_results = self._evaluate(pipeline, X, y)
-                score = evaluation_results['cv_score_mean']
-                score_to_minimize = -score if self.objective.greater_is_better else score
-                current_batch_pipeline_scores.append(score_to_minimize)
-                self._automl_algorithm.add_result(score_to_minimize, pipeline)
+            current_batch_size = len(current_batch_pipelines)
+            current_batch_pipeline_scores = self._evaluate_pipelines(current_batch_pipelines, X, y, search_iteration_plot=search_iteration_plot)
 
-                if search_iteration_plot:
-                    search_iteration_plot.update()
+            if search_iteration_plot:
```

Comment:
We call `update()` here and in `self._evaluate_pipelines`. I don't think we need both?

Reply:
Originally I wanted to be able to make sure that all of the results were plotted if the search was interrupted early. However, I did just test it and the one instance in `self._evaluate_pipelines` was sufficient so I'll remove the redundant one in `search()`.